src/economy.cpp

```diff
@@ -609,6 +609,7 @@ static void CompanyCheckBankrupt(Company *c)
 				 * is no THE-END, otherwise mark the client as spectator to make sure
 				 * he/she is no long in control of this company. However... when you
 				 * join another company (cheat) the "unowned" company can bankrupt. */
+				c->months_of_bankruptcy--;
```

Comment:
Couldn't we just change the variable type to a short or int? This seems a workaround.